node/src/blockchain/test_utils.rs

```diff
@@ -186,175 +161,13 @@ impl ReceiptResponseBuilder {
 
         let rpc_response = RpcResponse {
```

Comment:
I cannot comment in a higher place so I'm doing here:
I'm surprised that you've preserved the ReceiptResponseBuilder. Does it also mean we still fetch the entire TransactionReceipt during the retrieve_transactions process? We discussed and concluded that this structure is unbearably excessive compared our needs (just a little fraction).

Reply:
If we could we would really like to keep `ReceiptResponseBuilder`.
Being only test code I hope this isn't to much of a problem.
I predict in the future with us using MBCS more that `ReceiptResponseBuilder` will come in handy.

Reply:
Ehhh...that wasn't my point...but I think I didn't make myself clear. 

I just meant that I've been saying for a while that there will be time when we will have to get rid of our dependency on Web3. We don't want to work with such sizable structures as TransactionReceipt (from Web3) and therefore I hoped you would boil it down to our own and much smaller structure which could contain just that information we want. I'm certain it could be done while it would most likely also nicely work for any new chain added. However this isn't cheap in terms of time spent on refactoring it. So I will probably give you a green now. 

I myself wouldn't say it's completed if I were the author. I'd try to get rid of this boilerplate. If you don't feel up to this, let's throw it behind our back, but a TODO close to these problematic Web3 structures would be an advantage to have in the future. Please check the interface for any bloated structures fetched by an RPC code out of which we need just a little fragment of its whole. Write corresponding TODOS best at the trait declaration for an interface.

Reply:
Ah I see, Sorry I must of misunderstood you.
We created a new Struct `TxReceipt` which is a striped down version.
I believe this is what you are asking for.